### PR TITLE
Change the window titlebar to show the joined server

### PR DESCRIPTION
**Pull request by VasilisThePikachu**
## About the PR

The games window title will now update with the hostname of the server. Updates with hostname changes, and can optionally be disabled by the server with the ``game.hostname_in_titlebar`` cvar (in case the server updates the manifest.yml to show the server name instead and does not have other server to warent having their server name on the title bar)

## Why / Balance

Makes the life of admins who have multiple game server joined easier by being able to find each server from the titlebar

Makes it easier for players to identify what server they joined if they forgot.

Looks cool.

## Media

image

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl: Myra
- add: The game's title bar window will display the name of the server you have joined (unless disabled).

### CODE REVIEW

File: Content.Client/GameTicking/Managers/TitleWindowManager.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        _cfg.OnValueChanged(CCVars.HostnameInTitlebar, OnHostnameTitleChange, true);
+    }
+
+    public void Shutdown()
+    {
+        _cfg.UnsubValueChanged(CVars.GameHostName, OnHostnameChange);
+        _cfg.UnsubValueChanged(CCVars.HostnameInTitlebar, OnHostnameTitleChange);
+    }
+
+    // This may use the last joined server temporarily until the CCVars of the joining server are received.
```
Comment (PJB3005) [inline Content.Client/GameTicking/Managers/TitleWindowManager.cs:27]: This should probably be cleared on disconnect.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: Content.Shared/CCVar/CCVars.cs [hunk 2]
```diff
@@ -436,6 +436,12 @@ public static readonly CVarDef<bool>
         public static readonly CVarDef<float> GameEntityMenuLookup =
             CVarDef.Create("game.entity_menu_lookup", 0.25f, CVar.CLIENTONLY | CVar.ARCHIVE);
 
+        /// <summary>
+        /// Should the clients window show the server hostname in the title?
+        /// </summary>
+        public static readonly CVarDef<bool> HostnameInTitlebar =
```
Comment (PJB3005) [inline Content.Shared/CCVar/CCVars.cs:442]: This needs to be called `GameHostnameInTitlebar`.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: Content.Client/GameTicking/Managers/TitleWindowManager.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            _clyde.SetWindowTitle(hostname + " - " + defaultWindowTitle);
+        else
+            _clyde.SetWindowTitle(defaultWindowTitle);
+    }
+
+    // Clients by default assume game.hostname_in_titlebar is true (and also that the server hostname is MyServer)
+    // but we need to clear it as soon as we join and actually receive the CCVar.
+    // This will ensure we rerun OnHostnameChange and set the correct title bar name.
+    // We could also initialize this later but uhh I was told to make this a manager and put it into entrypoint.
+    private void OnHostnameTitleChange(bool colonthree)
```
[the remark this answers is not in the document]
Comment (VasilisThePikachu) [reply Content.Client/GameTicking/Managers/TitleWindowManager.cs:52]: :3

**Comment by VasilisThePikachu** [comment]
Tests are failing cause its trying to pull engine changes and it gets a conflict on the release file... even removing it does not help

File: Content.Client/GameTicking/Managers/TitleWindowManager.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+        _client.RunLevelChanged += OnRunLevelChangedChange;
+    }
+
+    public void Shutdown()
+    {
+        _cfg.UnsubValueChanged(CVars.GameHostName, OnHostnameChange);
+        _cfg.UnsubValueChanged(CCVars.GameHostnameInTitlebar, OnHostnameTitleChange);
+    }
+
```
Comment (metalgearsloth) [inline Content.Client/GameTicking/Managers/TitleWindowManager.cs:29]: Initialize is only called once upon game open (as it runs out of sim) so this doesn't really matter but it's also not hurting anyone so.

**Review by metalgearsloth — APPROVED** [review]

File: Content.Client/GameTicking/Managers/TitleWindowManager.cs (hunk 4 above)
Comment (VasilisThePikachu) [reply Content.Client/GameTicking/Managers/TitleWindowManager.cs:29]: I dunno I was told by nik to add it to not suffer someone's wrath

Comment (PJB3005) [reply Content.Client/GameTicking/Managers/TitleWindowManager.cs:29]: It doesn't hurt anything to have it so it's fine.

**Pull request merged**